WPMedia/arc-themes-components
# tmedia 689 autoplay carousel

## Ticket

- [TMEDIA-689](https://arcpublishing.atlassian.net/browse/TMEDIA-689) sub-task of https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TMEDIA-689&assignee=5e387d6a1b1d910e5dfd7a9b

## Description

Add autoplay and custom button functionality

## Acceptance Criteria

- Autoplay functionality opt-in and -out on arrival to the page of the carousel
- Feature is accessible by detecting reduced motion 

## Test Steps

1. Checkout branch - `git checkout TMEDIA-689-autoplay-gallery`
2. Run Storybook `npm run storybook`
3. Look at the storybook for autoplay functionality 
![Screen Shot 2022-04-18 at 17 00 03](https://user-images.githubusercontent.com/5950956/163884389-9a02bef9-ea62-4b94-981b-a2302634a26a.png)
4. Notice "Start Autoplay" on first arrival to a story with `enableAutoplay`
5. Click `Start Autoplay`
6. Wait 4 seconds (default from engine theme sdk). Then you will go to the next slide. You will also see the "Stop Autoplay" available 
![Screen Shot 2022-04-18 at 17 01 32](https://user-images.githubusercontent.com/5950956/163884574-1cc5da8e-1414-4eae-a0ac-c52e70a79a17.png)
8. Go to the custom button story example to see the buttons with our own custom styling work as before 
![Screen Shot 2022-04-18 at 17 02 44](https://user-images.githubusercontent.com/5950956/163884738-e1318982-f16e-4ae1-9c25-c48160b7c8a9.png)

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



-----

## Notes 

- Checking for reduced motion per accessibility requirement https://www.w3.org/TR/wai-aria-practices-1.2/examples/carousel/carousel-1-prev-next.html

## Questions 


- The engine theme sdk does not let you set the interval duration. I didn't implement this feature but set the 4000 from engine theme sdk interval along with its same logic. Is this ok to leave out? 
- Should we take in autoplay phrases as well as custom buttons? I think that the buttons should be enough customization. Thoughts? This follows the same pattern as the next and previous and fullscreen buttons. 

## Todos 

- [x] Take in autoplay components in storybook
- [x] Include autoplay custom buttons in proptypes 
- [x] Describe feature in carousel docs
- [x] Add tests. Include mocked reduced motion 
- [x] Test clicking to initiate the autoplay
